File: phase1/chapters/searching_algorithms/fibonacci_search/implementations/fibonacci_search.js
const fibMonaccianSearch = (arr, x, n) => {
  let fibMMm2 = 0; // (m-2)'th Fibonacci No.
  let fibMMm1 = 1; // (m-1)'th Fibonacci No.
  let fibM = fibMMm2 + fibMMm1; // m'th Fibonacci

  while (fibM < n) {
    fibMMm2 = fibMMm1;
    fibMMm1 = fibM;
    fibM = fibMMm2 + fibMMm1;
  }

  let offset = -1;

  while (fibM > 1) {
    let i = Math.min(offset + fibMMm2, n - 1);

    if (arr[i] < x) {
      fibM = fibMMm1;
      fibMMm1 = fibMMm2;
      fibMMm2 = fibM - fibMMm1;
      offset = i;
    } else if (arr[i] > x) {
      fibM = fibMMm2;
      fibMMm1 = fibMMm1 - fibMMm2;
      fibMMm2 = fibM - fibMMm1;
    } else return i;
  }

  if (fibMMm1 && arr[n - 1] == x) {
    return n - 1;
  }

  return -1;
};

let arr = [10, 22, 35, 40, 45, 50, 80, 82, 85, 90, 100, 235];
let n = arr.length;
let x = 235;
let ind = fibMonaccianSearch(arr, x, n);
if (ind >= 0) {
  console.log("Found at index: " + ind);
} else {
  console.log(x + " isn't present in the array");
}

File: phase1/chapters/searching_algorithms/fibonacci_search/implementations/fibonacci_search.cpp
#include <iostream>
using namespace std;

int min(int x, int y)
{
    return (x <= y) ? x : y;
}

int fibonacci_search(int arr[], int x, int n)
{
    int fbM2 = 0;          // (m-2)'th Fibonacci number
    int fbM1 = 1;          // (m-1)'th Fibonacci number
    int fbM = fbM2 + fbM1; // m'th Fibonacci
    int offset = -1;
    while (fbM < n)
    {
        fbM2 = fbM1;
        fbM1 = fbM;
        fbM = fbM2 + fbM1;
    }
    while (fbM > 1)
    {
        int i = min(offset + fbM2, n - 1);
        if (arr[i] < x)
        {
            fbM = fbM1;
            fbM1 = fbM2;
            fbM2 = fbM - fbM1;
            offset = i;
        }
        else if (arr[i] > x)
        {
            fbM = fbM2;
            fbM1 = fbM1 - fbM2;
            fbM2 = fbM - fbM1;
        }
        else
            return i;
    }
    if (fbM1 && arr[offset + 1] == x)
        return offset + 1;
    return -1;
}

int main()
{
    int n;
    cout << "\nEnter the number of elements in array which should be less than 10";
    cin >> n;
    int arr[10];
    cout << "Enter elements in array";
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    int n = sizeof(arr) / sizeof(arr[0]);
    int x;
    cout << "\nEnter element to be searched :";
    cin >> x;
    cout << "Found at index:" << fibonacci_search(arr, x, n);
    return 0;
}